src/state_manager/mod.rs

```diff
@@ -979,6 +980,30 @@ where
         Ok(out)
     }
 
+    /// Retrieves miner faults.
+    pub fn miner_faults(
+        self: &Arc<Self>,
+        addr: &Address,
+        ts: &Arc<Tipset>,
+    ) -> Result<BitField, Error> {
+        let actor = self
+            .get_actor(addr, *ts.parent_state())?
+            .ok_or_else(|| Error::State("Miner actor not found".to_string()))?;
+
+        let state = miner::State::load(self.blockstore(), actor.code, actor.state)?;
+
+        let mut faults = Vec::new();
+
+        state.for_each_deadline(&self.chain_config.policy, self.blockstore(), |_, part| {
```

Comment:
I would prefer more explicit name here, like `partition`.

Reply:
It's not partition. It's `State`.

Reply:
Ah, you are referring to `part`?

Reply:
Yes I'm referring to `part`.

Reply:
Looks like `part` should be `deadline` and `dl` should be `partition`.